src/pages/index.js

```diff
@@ -92,26 +149,98 @@ function fillProfileForm() {
 
 profileEditButton.addEventListener("click", () => {
   formValidators["profile-form"].resetValidation();
+  popupWithForms["edit-modal"].resetButtonText();
   popupWithForms["edit-modal"].open();
   fillProfileForm();
 });
 
+profileAvatar.addEventListener("mouseover", () => {
+  avatarEditButton.classList.add("profile__avatar-button_visible");
+});
+
+avatarEditButton.addEventListener("click", () => {
+  //reset the validation before opening it
+  formValidators["avatar-form"].resetValidation();
+  popupWithForms["avatar-modal"].resetButtonText();
+  popupWithForms["avatar-modal"].open();
+});
+
+const updateProfilePicture = (inputValues) => {
+  api
+    .updateProfilePicture(inputValues["avatar-link"])
+    .then((res) => {
+      profileInfo.updateUserAvatar(res["avatar"]);
+      popupWithForms["avatar-modal"].changeButtonText();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+  //retrieveProfileInfo();
+  /*api
+    .loadUserInfo()
+    .then((profileData) => {
+      popupWithForms["avatar-modal"].changeButtonText();
+      profileInfo.updateUserAvatar(profileData["avatar"]);
+    })
+    .catch((err) => {
+      console.error(err);
+    });*/
+  popupWithForms["avatar-modal"].close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**